pkg/asset/tls.go

```diff
@@ -98,18 +109,55 @@ func newAPIKeyAndCert(caCert *x509.Certificate, caPrivKey *rsa.PrivateKey, altNa
 	return key, cert, err
 }
 
-func newKubeletKeyAndCert(caCert *x509.Certificate, caPrivKey *rsa.PrivateKey) (*rsa.PrivateKey, *x509.Certificate, error) {
+func newAdminKeyAndCert(caCert *x509.Certificate, caPrivKey *rsa.PrivateKey) (*rsa.PrivateKey, *x509.Certificate, error) {
 	key, err := tlsutil.NewPrivateKey()
 	if err != nil {
 		return nil, nil, err
 	}
 	config := tlsutil.CertConfig{
-		CommonName:   "kubelet",
-		Organization: []string{"kube-node"},
+		CommonName:   "cluster-admin",
+		Organization: []string{"system:masters"},
 	}
 	cert, err := tlsutil.NewSignedCertificate(config, key, caCert, caPrivKey)
 	if err != nil {
 		return nil, nil, err
 	}
 	return key, cert, err
 }
+
+// newBootstrapAuthTokenFile creates a static token file, containing a single
+// randomly-generated bearer token mapped to the user and group
+// 'kubelet-bootstrap'.
+//
+// This token file is meant to be used by kubelet to bootstrap its TLS
+// certificate as described on k8s.io/docs/admin/kubelet-tls-bootstrapping.
+func newBootstrapAuthTokenFile() ([]byte, error) {
+	// We must create at least 128 bits of entropy from a secure source.
+	bearer, err := generateRandomString(16)
+	if err != nil {
+		return nil, err
+	}
+
+	token := fmt.Sprintf("%s,kubelet-bootstrap,10001,\"system:kubelet-bootstrap\"\n", bearer)
+	return []byte(token), nil
+}
+
+// getBootstrapAuthTokenFromFile retrieves the first bearer token from the
+// given data of a static token file.
+func getBootstrapAuthTokenFromFile(data []byte) (string, error) {
```

Comment:
This doesn't actually get anything from a file. Maybe `parseBootstrapAuthToken`?